### PR TITLE
Storage feature for virtual and hardware servers

### PR DESCRIPTION
**Pull request by FernandoOjeda**
Storage feature for virtual and hardware servers https://github.com/softlayer/softlayer-python/issues/1241.

### CODE REVIEW

File: SoftLayer/CLI/hardware/storage.py [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+                                 title="File Storage Details")
+    for nas in nas_storage_data:
+        table_nas.add_row([nas['username'], nas['capacityGb'],
+                           nas['serviceResourceBackendIpAddress'],
+                           nas['allowedHardware'][0]['datacenter']['longName'],
+                           nas.get('notes', None)])
+
+    table_hard_drives = formatting.Table(['Type', 'Name', 'Capacity', 'Serial #'], title="Other storage details")
+    for drives in hard_drives:
+        table_hard_drives.add_row([drives['hardwareComponentModel']['hardwareGenericComponentModel']
```
Comment (allmightyspiff) [inline SoftLayer/CLI/hardware/storage.py:48]: To make this line read easier, assign type/name/capacity/serial to variables, and then just have those variables in the add_row function.

**Comment by FernandoOjeda** [comment]
About the local disk storage do you mean to these?
Virtual Server:
image

image


Hardware Server:
image

It seems that hardware server storage in this part only shows the hard drives and there is not a relational datatype for blockdevices the same that the VS.

**Comment by allmightyspiff** [comment]
Yes, I want those to show up on `slcli hardware|vs detail`

for hardware, isn't `hard_drives = hardware.get_hard_drives(hardware_id)` getting the drives like in your screenshot?

**Comment by allmightyspiff** [comment]
Something like this:

```
$ slcli hw detail 943379
id          943379
guid        31e20db3-d7b5-4492-944f-e2215670b856
hostname    host0.vmware
domain      chechu.com
fqdn        host0.vmware.chechu.com
status      ACTIVE
datacenter  fra02
cores       12
memory      262144
drives      
    Seagate Enterpries  12000GB   zxc9d78s
    Seagate Enterpries  12000GB   zxc9d722
public_ip   158.177.97.206
private_ip  10.85.94.142
ipmi_ip     10.85.94.146
os          VSphere
os_version  6.5.0u1
created     2017-12-20T12:21:42-06:00
owner       schaparro
vlans       PRIVATE  1202  2206791
            PUBLIC   1126  2206789
Bandwidth   Private  4420.03379  13813.32856  N/A
            Public   1647.15639  26.74248     500
tags        tape tool: chechu-testing, tape tool: vmware-demo-lab-chechu
```

**Comment by FernandoOjeda** [comment]
ok sure, I will add the local disks information for both slcli hardware|vs detail, the same that your example.

File: SoftLayer/CLI/hardware/storage.py (hunk 1 above)
Comment (FernandoOjeda) [reply SoftLayer/CLI/hardware/storage.py]: Done.

**Pull request merged**